src/hip_memory.cpp

```diff
@@ -1901,6 +1897,22 @@ hipError_t hipFree(void* ptr) {
         am_status_t status = hc::am_memtracker_getinfo(&amPointerInfo, ptr);
         if (status == AM_SUCCESS) {
             if (amPointerInfo._hostPointer == NULL) {
+                if (amPointerInfo._appId != -1) {
```

Comment:
Do we still need this check? Since we are synchronizing all devices for now.

Reply:
I thought this was differentiating between the case where the pointer is owned and not shared versus when it is owned but shared.  Perhaps I misunderstood HCC pointer management and what the _appId field is used for.  I don't want hipFree() to become a performance bottleneck if we can help it.

Reply:
_appId= -1 is used for host allocations. I don't think hipFree should be performance critical code.

Reply:
Synchronizing all devices may help avoiding future mGPU related issues when devices are sharing pointers and freeing across devices. Does blanket all device synchronization resolve 183319?